test/consensus/transaction_test.cpp

```diff
@@ -0,0 +1,79 @@
+/*
+Copyright Soramitsu Co., Ltd. 2016 All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+#include "../../core/consensus/sumeragi.hpp"
+
+#include "../../core/consensus/connection/connection.hpp"
+#include "../../core/model/commands/transfer.hpp"
+#include "../../core/model/commands/add.hpp"
+
+#include <iostream>
+#include <string>
+#include <vector>
+#include <memory>
+#include <thread>
+
+#include "../../core/service/json_parse_with_json_nlohman.hpp"
+
+#include "../../core/service/peer_service.hpp"
+#include "../../core/crypto/hash.hpp"
+
+template<typename T>
+using Transaction = transaction::Transaction<T>;
+template<typename T>
+using ConsensusEvent = event::ConsensusEvent<T>;
+template<typename T>
+using Add = command::Add<T>;
+template<typename T>
+using Transfer = command::Transfer<T>;
+
+void setAwkTimer(int const sleepMillisecs, std::function<void(void)> const action) {
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Function parameter 'action' should be passed by reference.](https://www.codacy.com/app/soramitsu/iroha/file/4478128098/issues/source?bid=3712491&fileBranchId=3923596#l43)